File: carpool-service/src/main/java/com/tompang/carpool/carpool_service/command/domain/ride_request/event/RideRequestAcceptedEvent.java
package com.tompang.carpool.carpool_service.command.domain.ride_request.event;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.tompang.carpool.carpool_service.common.DomainTopics;

/**
 * Event when ride request is accepted by a carpool, assigning them together.
 */
public class RideRequestAcceptedEvent implements RideRequestEvent {

    public final String carpoolId;
    public final String requestId;

    @JsonCreator
    public RideRequestAcceptedEvent(
        @JsonProperty("requestId") String requestId,
        @JsonProperty("carpoolId") String carpoolId
    ) {
        this.requestId = requestId;
        this.carpoolId = carpoolId;
    }

    @Override
    public String topicName() {
        return DomainTopics.RideRequest.REQUEST_ACCEPTED;
    }

}

File: notification-service/src/message/event/ride-request/ride-request-accepted.event.ts
import { DomainEvent } from "..";

export class RideRequestAcceptedEvent implements DomainEvent {
  requestId: string;
  carpoolId: string;
  riderId: string;

  constructor(message: any) {
    this.requestId = message.requestId;
    this.carpoolId = message.carpoolId;
    this.riderId = message.riderId;
  }

  getMessage(): string {
    return `A carpool has accepted your ride request`;
  }
  getTargetUserId(): string {
    return this.riderId;
  }
}